### PR TITLE
feat: Implement delegation validation rules in domain model

### PR DESCRIPTION
**Pull request by FirestarJes**

### CODE REVIEW

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Delegations/MessageDelegation.cs [hunk 1]
```diff
@@ -69,12 +92,26 @@ public void StopDelegation(DelegationPeriod existingPeriod, Instant stopsAt)
 
         if (!_delegations.Remove(existingPeriod))
         {
-            throw new InvalidOperationException("Provided existing delegation period was not in collection.");
+            throw new ValidationException("Provided existing delegation period was not in collection.")
```
Comment (defectiveAi) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Delegations/MessageDelegation.cs:95]: This one should still be InvalidOpException - this is a coding bug, if happens.

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Delegations/MessageDelegation.cs [hunk 2]
```diff
@@ -57,8 +74,14 @@ public void DelegateTo(ActorId delegatedTo, GridAreaId gridAreaId, Instant start
     {
         var delegationPeriod = new DelegationPeriod(delegatedTo, gridAreaId, startsAt, stopsAt);
 
-        // TODO: Rule (A/Fra, MessageType) skal være unik i perioden i netområde.
-        // TODO: Rule Denne regel gælder ikke, hvis ExpiresAt <= StartsAt.
+        // Rule There can't be any overlap between delegation period on a given grid area
```
Comment (defectiveAi) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Delegations/MessageDelegation.cs:77]: Remove both comments, or combine and move them above IsThereDelegationPeriodOverlap method.

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Delegations/MessageDelegation.cs (hunk 1 above)
Comment (FirestarJes) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Delegations/MessageDelegation.cs]: Fixed

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Delegations/MessageDelegation.cs (hunk 2 above)
Comment (FirestarJes) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Delegations/MessageDelegation.cs]: Both comments removed

**Review by defectiveAi — APPROVED** [review]

**Pull request merged**